src/components/PostDetails.tsx

```diff
@@ -1,110 +1,62 @@
+  const { setSelectedPost } = useAppSelector(state => state.author);
+  const {
+    comments,
+    hasError,
+    loaded,
+    visible,
+  } = useAppSelector(state => state.comments);
+  const dispatch = useAppDispatch();
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
-    setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+    if (setSelectedPost) {
+      dispatch(commentActions.init(setSelectedPost.id));
     }
-  };
+  }
 
   useEffect(() => {
     loadComments();
-  }, []);
-
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+  }, [setSelectedPost?.id]);
 
   const addComment = async ({ name, email, body }: CommentData) => {
-    try {
-      const newComment = await commentsApi.createComment({
+    if (setSelectedPost) {
+      const newComment = {
         name,
         email,
         body,
-        postId: post.id,
-      });
-
-      setComments(
-        currentComments => [...currentComments, newComment],
-      );
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
+        postId: setSelectedPost?.id,
+      };
+
+      dispatch(commentActions.addComment(newComment));
     }
   };
 
-  const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    setComments(
-      currentComments => currentComments.filter(
-        comment => comment.id !== commentId,
-      ),
-    );
-
-    await commentsApi.deleteComment(commentId);
+  const deleteComment = (commentId: number) => {
+    dispatch(commentActions.removeComment(commentId));
+    dispatch(commentActions.actions.setCommentId(commentId));
   };
 
   return (
     <div className="content" data-cy="PostDetails">
       <div className="block">
         <h2 data-cy="PostTitle">
-          {`#${post.id}: ${post.title}`}
+          {`#${setSelectedPost?.id}: ${setSelectedPost?.title}`}
         </h2>
 
         <p data-cy="PostBody">
-          {post.body}
+          {setSelectedPost?.body}
```

Comment:
Use destructuring for objects for better readability

Reply:
I have in this file the same simple names, for instance body, title. I can't do destructuring.